src/main/webapp/app/code-editor/code-editor.component.ts

```diff
@@ -151,18 +155,27 @@ export class CodeEditorComponent implements OnInit, OnChanges, OnDestroy {
         this.latestResult = $event.newResult;
     }
 
+    /**
+     * Check if the received build logs are recent and format them for use in the ace-editor
+     * @param buildLogs
+     */
     updateLatestBuildLogs(buildLogs: BuildLogEntry[]) {
-        this.buildLogErrors = buildLogs
-            .map(({ log }) => log && log.match(this.errorLogRegex))
-            .filter(log => !!log && log.length === 6 && log[1] === 'ERROR')
-            .map(([, , fileName, row, column, text]) => ({
-                type: 'error',
-                fileName,
-                row: Math.max(parseInt(row, 10) - 1, 0),
-                column: Math.max(parseInt(column, 10) - 1, 0),
-                text: safeUnescape(text)
-            }))
-            .reduce((acc, { fileName, ...rest }) => ({ ...acc, [fileName]: [...(acc[fileName] || []), rest] }), {});
+        const ts = buildLogs.length ? Date.parse(buildLogs[0].time) : 0;
+        if (!this.buildLogErrors || ts > this.buildLogErrors.ts) {
+            const errors = buildLogs
+                .map(({ log, time }) => log && { log: log.match(this.errorLogRegex), time })
+                .filter(({ log }) => !!log && log.length === 6 && log[1] === 'ERROR')
+                .map(({ log: [, , fileName, row, column, text], time }) => ({
+                    type: 'error',
+                    fileName,
+                    row: Math.max(parseInt(row, 10) - 1, 0),
+                    column: Math.max(parseInt(column, 10) - 1, 0),
+                    text: safeUnescape(text),
+                    ts: Date.parse(time)
+                }))
+                .reduce((acc, { fileName, ...rest }) => ({ ...acc, [fileName]: [...(acc[fileName] || []), rest] }), {});
```

Comment:
what does acc stands for?

Reply:
It's short for accumulator, this term is also in used in official documentation and an informal javascript convention (e.g. https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce). There are other conventions as "prev", but none of them make much sense to me.

Reply:
I think accumulator is too generic. Can we find a better term here? If not, then please put in the full name 'accumulator' otherwise someone might think acc stands for account or something else.

Also add a comment what you want to achieve with reduce ;-) Otherwise the code is hard to understand. I could not tell from looking at it, what exactly you are doing here

Reply:
I'll change it and add some comments.

Reply:
Done.